test/unit/tasks/cms/cms_test.rb

```diff
@@ -2,24 +2,56 @@
 
 module Tasks
   class CmsTest < ActiveSupport::TestCase
+    class CmsTestFixPaginationHref < CmsTest
+      def setup
+        @page = FactoryBot.build(:cms_partial, system_name: 'shared/pagination',
+                                 published: '<p>Title</p><a href="{{ part.link }}">Supertramp</a>')
 
-    def setup
-      @page = FactoryBot.build(:cms_partial, system_name: 'shared/pagination',
-        published: '<p>Title</p><a href="{{ part.link }}">Supertramp</a>')
+        @page.save(validate: false)
+      end
 
-      @page.save(validate: false)
+      def test_fix_pagination_href
+        assert_equal true,  @page.published.include?('{{ part.link }}')
+        assert_equal false, @page.published.include?('{{ part.url }}')
+
+        execute_rake_task 'cms/cms.rake', 'cms:fix:pagination_href'
+
+        @page.reload
+
+        assert_equal true,  @page.published.include?('{{ part.url }}')
+        assert_equal false, @page.published.include?('{{ part.link }}')
+      end
     end
 
-    def test_fix_pagination_href
-      assert_equal true,  @page.published.include?('{{ part.link }}')
-      assert_equal false, @page.published.include?('{{ part.url }}')
+    class CmsTestFixSectionEmptyTitles < CmsTest
+      def setup
+        @provider = FactoryBot.create(:provider_account)
+      end
+
+      [nil, ""].each do |empty_val|
+        define_method("test_fix_section_empty_titles_title_#{empty_val.nil? ? 'nil' : 'empty'}") do
```

Comment:
I think this approach makes it harder to read, and therefore to maintain.

Reply:
Readability vs. Not repeating code. I wrote separated tests at first and then merged them suggested by @akostadinov. I think the current state is better but I don't have a strong opinion, so if you convince Aleks, I'm fine.

Reply:
Readability can be improved by using `test "..."` syntax instead of `define_method`. Or by defining a test and then just loop through the two values inside it. 

```rb
[nil, ""].each do |empty_val| do |value|
  test "section titles are fixed for #{value.inspect}" do
    # actual test code
  end
end
```

```rb
  test "section empty titles are fixed" do
    [nil, ""].each do |empty_val|
      # actual test code
    end
  end
```

Reply:
I'd vote for the 2nd one @akostadinov suggested.

Reply:
I pushed the first option, because the second would create only one test and run the rake task twice without a DB rollback between them